### DISCUSSION TITLE
[Feature Request] Run TypeScript check and ESLint simultaneously

### DISCUSSION THREAD

**Discussion by SukkaW** [vercel/next.js#37088] [Ideas]
### Describe the feature you'd like to request

Currently, Next.js runs TypeScript type check and ESLint one by one:

https://github.com/vercel/next.js/blob/b122178ead5f6a50e0ea543e1e806de93bde23af/packages/next/build/index.ts#L245-L255
https://github.com/vercel/next.js/blob/b122178ead5f6a50e0ea543e1e806de93bde23af/packages/next/build/index.ts#L278-L291

It takes about 12s to finish type check and 10s to finish ESLint in my project. If we can type check and lint simultaneously, we might be able to reduce the build time.

### Describe the solution you'd like

Currently, Next.js already runs ESLint in a separate worker:
https://github.com/vercel/next.js/blob/b122178ead5f6a50e0ea543e1e806de93bde23af/packages/next/lib/verifyAndLint.ts#L20-L23

However, the typescript check is still run in the main thread:
https://github.com/vercel/next.js/blob/b122178ead5f6a50e0ea543e1e806de93bde23af/packages/next/lib/verifyTypeScriptSetup.ts#L77-L82

### Describe alternatives you've considered

N/A

**Comment by ijjk** [comment]
Hi, this sounds like a good optimization, feel free to submit a PR updating this!

**Reply by SukkaW** [reply]
Implemented in https://github.com/vercel/next.js/pull/37105.
